src/Controllers/GagePageController.ts

```diff
@@ -536,13 +581,300 @@ module StreamStats.Controllers {
                 }
             }
         }
+
+        // Gage Analysis Plots methods
+
+        public getGagePlot() {
+            //instantiate gage
+            this.gage = new GageInfo(this.modalService.modalOptions.siteid);
+            //below: how to get the data from API and setting it in controller
+            const url = configuration.baseurls.GageStatsServices + configuration.queryparams.GageStatsServicesStations + this.gage.code;
+            const request: WiM.Services.Helpers.RequestInfo = new WiM.Services.Helpers.RequestInfo(url, true, WiM.Services.Helpers.methodType.GET, 'json');
+
+            this.Execute(request).then(
+                (response: any) => {
+                    this.gage = response.data;
+                    this.getPeakInfo();
+                    this.getFloodFreq();
```

Comment:
`this.getFloodFreq()` is called on this line and also on line 646. Same with `this.getDailyFlow()`-- it is called here and on line 672. I think these functions are being run twice, when they only needs to run once? If so, you probably want to omit these two lines (598 and 599).